### PR TITLE
Gateway conformance

### LINKED ISSUE #447: Run conformance test in CI
the gateway conformance test suite should be run / pass in CI against the L1 node

### PR DESCRIPTION
**Pull request by guanzo**
Fixes #447. Runs the L1 through the [gateway conformance](https://github.com/ipfs/gateway-conformance) tests for specs: `trustless-block-gateway`, `trustless-car-gateway`

* Configures lassie to fetch test cids from a localhost kubo node.
* Adds njs module to set the `content-disposition` header
* All tests pass except for range requests.

A bunch of other commits got pulled in somehow >.<, sorry

### CODE REVIEW

File: container/nginx/njs/ipfs-response.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+function setHeaders(req) {
+  const format = req.variables.format_final;
+  const filename = req.args.filename;
+
+  const pathSegments = req.uri.split("/");
+  const cid = pathSegments[2];
+  const filePathSegments = pathSegments.slice(3);
+  const lastSegment = filePathSegments[filePathSegments.length - 1];
+
+  let name;
```
Comment (joaosa) [inline container/nginx/njs/ipfs-response.js:10]: Is it possible for us to enter this filter and return an undefined value for `name`?

File: .github/workflows/gateway-conformance.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          specs: trustless-block-gateway,trustless-car-gateway
+          json: output.json
+          xml: output.xml
+          html: output.html
+          markdown: output.md
+          args: -skip 'TestGatewayCar/GET_response_for_application/vnd.ipld.car/Header_Content-Length'
+
+      # 7. Upload the results
+      - name: Upload Markdown summary
+        if: failure() || success()
```
Comment (joaosa) [inline .github/workflows/gateway-conformance.yml:100]: Isn't this `if` redundant or is there another possible state?

File: container/nginx/njs/ipfs-response.js (hunk 1 above)
Comment (guanzo) [reply container/nginx/njs/ipfs-response.js:10]: ya, if there's no filename or format, the L1 will return a 501 and there'll be a blank value in the header. Addressed with https://github.com/filecoin-saturn/L1-node/pull/449/commits/c8e8eab73edfaaa0d08af605e106f9d3c2616a0b

File: .github/workflows/gateway-conformance.yml (hunk 2 above)
Comment (guanzo) [reply .github/workflows/gateway-conformance.yml:100]: https://stackoverflow.com/a/58859404

Comment (DiegoRBaquero) [reply .github/workflows/gateway-conformance.yml:100]: Isn't not canceled (`'!cancelled()'`) more precise then?

**Comment by guanzo** [comment]
fixed commits

File: .github/workflows/gateway-conformance.yml (hunk 2 above)
Comment (guanzo) [reply .github/workflows/gateway-conformance.yml]: ya, updated

**Review by DiegoRBaquero — APPROVED** [review]

**Pull request merged**